src/lib/ngsiNotify/QueueStatistics.cpp

```diff
@@ -122,3 +130,38 @@ void QueueStatistics::incSentError()
 {
   __sync_fetch_and_add(&noOfNotificationsQueueSentError, 1);
 }
+/* ****************************************************************************
+*
+* getTimInQ -
+*/
+void QueueStatistics::getTimeInQ(char* buf, size_t bufLen)
+{
+  boost::mutex::scoped_lock lock(mtxTimeInQ);
```

Comment:
I understand that this is to lock the mutet until the current scope is destroyed (i.e. the getTimeInQ() function returns) so no explicit unlock statement is needed. Is that correct?

(This is specially interesting for functions which many exit point, we have several cases of them in mongoBackend and missing the unlock before `return` causes bug... one of them fixes few days ago: https://github.com/telefonicaid/fiware-orion/issues/1488)